src/hotspot/share/prims/jvmtiEnvBase.hpp

```diff
@@ -499,36 +489,44 @@ class JvmtiHandshake : public Handshake {
   static void execute(JvmtiUnitedHandshakeClosure* hs_cl, jthread target);
 };
 
-class SetForceEarlyReturn : public JvmtiHandshakeClosure {
+class SetForceEarlyReturn : public JvmtiUnitedHandshakeClosure {
 private:
   JvmtiThreadState* _state;
   jvalue _value;
   TosState _tos;
 public:
   SetForceEarlyReturn(JvmtiThreadState* state, jvalue value, TosState tos)
-    : JvmtiHandshakeClosure("SetForceEarlyReturn"),
+    : JvmtiUnitedHandshakeClosure("SetForceEarlyReturn"),
      _state(state),
      _value(value),
      _tos(tos) {}
+  void doit(Thread *target, bool self);
   void do_thread(Thread *target) {
-    doit(target, false /* self */);
+    doit(target, _self);
+  }
+  void do_vthread(Handle target_h) {
+    assert(_target_jt != nullptr, "sanity check");
```

Comment:
Better to test that target_h is same as _target_jt.

Reply:
Thanks.
The `_target_jt` is a `JavaThread*` while the `target_h` is a handle of thread oop.
Added the following assert: `assert(_target_jt->vthread() == target_h(), "sanity check");`